src/sentry/sentry_metrics/consumers/indexer/batch.py

```diff
@@ -161,26 +161,81 @@ def reconstruct_messages(
             used_tags.add(metric_name)
 
             new_tags: MutableMapping[str, int] = {}
+            exceeded_global_quotas = 0
+            exceeded_org_quotas = 0
+
             try:
                 for k, v in tags.items():
                     used_tags.update({k, v})
-                    new_tags[str(mapping[org_id][k])] = mapping[org_id][v]
+                    new_k = mapping[org_id][k]
+                    new_v = mapping[org_id][v]
+                    if new_k is None:
+                        metadata = bulk_record_meta.get(k)
+                        if (
+                            metadata
+                            and metadata.fetch_type_ext
+                            and metadata.fetch_type_ext.is_global
+                        ):
+                            exceeded_global_quotas += 1
+                        else:
+                            exceeded_org_quotas += 1
+                        continue
+
+                    if new_v is None:
+                        metadata = bulk_record_meta.get(v)
+                        if (
+                            metadata
+                            and metadata.fetch_type_ext
+                            and metadata.fetch_type_ext.is_global
+                        ):
+                            exceeded_global_quotas += 1
+                        else:
+                            exceeded_org_quotas += 1
+                        continue
+
+                    new_tags[str(new_k)] = new_v
             except KeyError:
                 logger.error("process_messages.key_error", extra={"tags": tags}, exc_info=True)
                 continue
 
+            if exceeded_org_quotas or exceeded_global_quotas:
```

Comment:
Please write a datadog metric here as well. The error is likely going to be dropped (we do not need sentry errors when a customer floods us) though the metric will still be useful